### PR TITLE
deps(go): use cosmos/ibc-go v6.3.0

### LINKED ISSUE #9205: Upgrade Ibc-go to address ASA-2024-007
## What is the Problem Being Solved?
[ASA-2024-007](https://github.com/cosmos/ibc-go/security/advisories/GHSA-j496-crgh-34mx) describes a vulnerability in the version of IBC used by the Agoric chain. There are no known ways to exploit this vulnerability on the Agoric chain. However, to ensure we're ready for rapid deployment of fixes, let's apply this change to the `master` branch of agoric-sdk.

## Description of the Design
Simple patch to bump version in https://github.com/Agoric/agoric-sdk/blob/master/golang/cosmos/go.mod 

## Security Considerations
This is a hardening measure

## Scaling Considerations
No known impact to scaling

## Test Plan
Please testing CI as well as using `/golang/cosmos/e2e_test`

## Upgrade Considerations
This should go out in upgrade16 regardless of whatever other Interchain Stack changes go out.

### PR DESCRIPTION
**Pull request by LuqiPan**
closes: #9205
refs: #9205

## Description

This PR updates ibc-go in response to https://github.com/cosmos/ibc-go/security/advisories/GHSA-j496-crgh-34mx

TODO
- [x] upgrade our runbook with a note on async channel version negotiation

### Security Considerations

This PR addresses https://github.com/cosmos/ibc-go/security/advisories/GHSA-j496-crgh-34mx

### Scaling Considerations

N/A

### Documentation Considerations

N/A

### Testing Considerations

Relying on our CI/CD to verify things as this is a simple dependency version bump

### Upgrade Considerations

N/A

### CODE REVIEW

File: golang/cosmos/go.mod [hunk 1]
```diff
@@ -192,9 +192,6 @@ replace (
 	// Pick up an IAVL race fix.
 	github.com/cosmos/iavl => github.com/cosmos/iavl v0.19.7
 
-	// Async version negotiation
-	github.com/cosmos/ibc-go/v6 => github.com/agoric-labs/ibc-go/v6 v6.2.1-alpha.agoric.3
```
Comment (gibson042) [inline golang/cosmos/go.mod:196]: @michaelfig should confirm, but looking at the [diff of v6.2.1-alpha.agoric.3 from upstream v6.2.1](https://github.com/cosmos/ibc-go/compare/v6.2.1...agoric-labs:ibc-go:v6.2.1-alpha.agoric.3#diff-b225d8b78960f1ed19ccabcbdcda5ff708075790adac8463fbb0a83a8a63b78a), I don't think we can move away from our fork. And there are hopefully tests in agoric-sdk that will fail if we try.

Comment (michaelfig) [reply golang/cosmos/go.mod:196]: > I don't think we can move away from our fork. And there are hopefully tests in agoric-sdk that will fail if we try.

@LuqiPan's changes in this PR remove the async version negotiation feature detection between our ibc-go fork and agoric-sdk, which allows the tests to pass again while preventing agoric-sdk from using the AsyncVersion code.

In a meeting some time ago, we had agreed not to use that semi-automatic feature detection, and instead would document in the interchain stack upgrade runbook the conditions under which we could set `AsyncVersions = true`, namely by performing some investigation into the ibc-go implementation we rely on before landing PRs like this one.

**Review by michaelfig — CHANGES_REQUESTED** [review]
Blocking until the interchain stack upgrade runbook describes the work needed to upgrade `AsyncVersions`.  I'd be happy to help you document this, @LuqiPan.

**Comment by LuqiPan** [comment]
Created https://github.com/Agoric/agoric-sdk/issues/9358 as reference issue, I am updating this PR and interchain stack upgrade runbook to reference this issue.

**Comment by LuqiPan** [comment]
Updated this PR and https://github.com/Agoric/agoric-sdk/wiki/Upgrading-the-Interchain-Stack (at the bottom) with #9358, should be ready for another look

File: golang/cosmos/go.mod [hunk 2]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/armon/go-metrics v0.4.1
 	github.com/cosmos/cosmos-sdk v0.46.16
 	github.com/cosmos/ibc-apps/middleware/packet-forward-middleware/v6 v6.1.2
-	github.com/cosmos/ibc-go/v6 v6.2.1
+	github.com/cosmos/ibc-go/v6 v6.3.0
```
Comment (michaelfig) [inline golang/cosmos/go.mod]: Upstream `ibc-go` is at v6.3.1 now.  Could you please change the dependency to that instead?

**Review by michaelfig — CHANGES_REQUESTED** [review]

File: golang/cosmos/go.mod (hunk 2 above)
Comment (LuqiPan) [reply golang/cosmos/go.mod]: Good call, just bumped to v6.3.1

**Review by michaelfig — APPROVED** [review]

**Comment by aj-agoric** [comment]
Moved back to code complete since bot should not have moved this to "done-mn validated"

**Pull request merged**